dev/list_failed_cross_version_tests.py

```diff
@@ -0,0 +1,61 @@
+"""
+A script to list failed jobs in the latest scheduled run of the `cross-version-tests` workflow
+using the GitHub Actions API.
+
+References:
+- https://docs.github.com/en/rest/reference/actions#list-workflow-runs
+- https://docs.github.com/en/rest/reference/actions#list-jobs-for-a-workflow-run
+"""
+import json
+import os
+import requests
+
+
+def fetch_failed_jobs(session, run_id):
+    failed_jobs = []
+    per_page = 100
+    page = 1
+    while True:
+        url = f"https://api.github.com/repos/mlflow/mlflow/actions/runs/{run_id}/jobs"
+        resp = session.get(url, params={"per_page": per_page, "page": page})
+        jobs = resp.json()["jobs"]
+        failed_jobs.extend(j for j in jobs if j["conclusion"] == "failure")
+        if len(jobs) < per_page:
+            break
+        page += 1
+
+    return failed_jobs
+
+
+class StrictSession(requests.Session):
+    """
+    A wrapper class for `requests.Session` to validate the request is successful.
+    """
+
+    def request(self, *args, **kwargs):
+        resp = super().request(*args, **kwargs)
+        resp.raise_for_status()
+        return resp
+
+
+def create_session():
+    session = StrictSession()
+    token = os.environ.get("GITHUB_TOKEN")
+    if token:
+        session.headers.update({"Authorization": f"token {token}"})
+    return session
+
+
+def main():
+    session = create_session()
+    url = (
+        "https://api.github.com/repos/mlflow/mlflow/actions/workflows/cross-version-tests.yml/runs"
+    )
+    resp = session.get(url, params={"event": "schedule"})
+    latest_run_id = resp.json()["workflow_runs"][0]["id"]
+    failed_jobs = fetch_failed_jobs(session, latest_run_id)
+    print(json.dumps(failed_jobs, indent=2))
```

Comment:
should the session be closed at the end to free the tcp connection? (might be a bit of a nit here)
Alternatively to manually specifying a close on the session, the `.get()`'s could be wrapped in a with clause at which the pool will be gc'd and the threads reallocated.
By default requests.Session uses urllib3's keep-alive setting == True. urllib3 will release a connection once the entire body has been read for pool reuse, but I've always just been in the habit of forcibly terminating a Session for fear of a rogue process just hanging around in a machine.

Reply:
Good point, let's use `with`.